azkaban-exec-server/src/main/java/azkaban/execapp/FlowPreparer.java

```diff
@@ -163,11 +164,10 @@ private String generateProjectDirName(final ProjectDirectoryMetadata proj) {
     return String.valueOf(proj.getProjectId()) + "." + String.valueOf(proj.getVersion());
   }
 
-  private File createTempDir(final ProjectDirectoryMetadata proj) {
+  private File createTempDir(final ProjectDirectoryMetadata proj) throws IOException {
     final String projectDir = generateProjectDirName(proj);
-    final File tempDir = new File(this.projectCacheDir,
-        "_temp." + projectDir + "." + System.currentTimeMillis());
```

Comment:
Please, please please don't reinvent filesystem primitives. You'll probably get it wrong. Like we did here.